packages/components/src/components/input/input.tsx

```diff
@@ -31,28 +69,28 @@ const Input = React.forwardRef(
             warn,
             data_testId,
             ...props
-        },
-        ref
+        }: TInputProps,
+        ref?: React.Ref<HTMLInputElement & HTMLTextAreaElement>
     ) => {
         const [counter, setCounter] = React.useState(0);
 
         React.useEffect(() => {
-            if (typeof initial_character_count === 'number') {
+            if (initial_character_count || initial_character_count === 0) {
                 setCounter(initial_character_count);
             }
         }, [initial_character_count]);
 
-        const changeHandler = e => {
+        const changeHandler = (e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
             let input_value = e.target.value;
             if (max_characters && input_value.length >= max_characters) {
                 input_value = input_value.slice(0, max_characters);
             }
             setCounter(input_value.length);
             e.target.value = input_value;
-            props.onChange(e);
+            props.onChange?.(e);
         };
 
-        const has_footer = has_character_counter || (hint && is_relative_hint);
+        const has_footer: boolean = !!has_character_counter || (!!hint && !!is_relative_hint);
```

Comment:
```suggestion
        const has_footer = !!has_character_counter || (!!hint && !!is_relative_hint);
```
@jim-deriv Could you please check if it will infer the boolean type without `: boolean` here? 🙂